lib/server/header_encoding.js

```diff
@@ -0,0 +1,94 @@
+"use strict";
+
+const zlib = require('zlib');
+
+/*
+const fs = require('fs');
+
+let readFile = async (filename) => { // into buffer
+  return new Promise(( resolve, reject ) => {
+    fs.readFile(filename, function(err, data) {
+      if ( err ) {
+        // console.log('error on readfile');
+        reject( err );
+      } else {
+        //console.log(JSON.parse(data));
+        resolve(new Buffer.from(data, 'utf8'));
+      }
+    });
+  });
+};
+*/
+
+let _gzipFile = async (inputBuffer) => {
+  return new Promise((resolve, reject) => {
+    zlib.gzip(inputBuffer, function( err, result ) {
+      if ( err ) { reject( err ); }
+      let value = result;
+      // fs.writeFile("fileHeaderInputGz.txt", value, (err) => {
+      //   if (err) { throw err; }
+      // });
+      resolve( value );
+    });
+  });
+};
+
+let _unzipFile = async (inputBuffer) => {
+  return new Promise((resolve, reject) => {
+    zlib.unzip(inputBuffer, function( err, result ) {
+      if ( err ) { reject( err ); }
+      let value = result;
+      // fs.writeFile("fileHeaderInputGzToTxt.txt", value, (err) => {
```

Comment:
There are blocks of commented text which I don't think we will use. If that is correct can you please delete them?